src/Components/StepThree/StepThree.js

```diff
@@ -89,33 +100,33 @@ const StepThree = ({prevStep, setData, data, submitForm, step, getRealDate}) =>
     }
 };
     useEffect(() => {
-        const stringMetalWorkSkills = inputs.metalworkSkills.toString();
+        const stringMetalWorkSkills = state.metalworkSkills.toString();
           setData((data) => ({
             ...data,
-            doesHaveAgricultureSkills: inputs.doesHaveAgricultureSkills,
-            agricultureSkills: inputs.agricultureSkills,
-            doesHaveMetalworkSkills: inputs.doesHaveMetalworkSkills,
+            doesHaveAgricultureSkills: state.doesHaveAgricultureSkills,
+            agricultureSkills: state.agricultureSkills,
+            doesHaveMetalworkSkills: state.doesHaveMetalworkSkills,
             metalworkSkills: stringMetalWorkSkills,
-            isConvicted: inputs.isConvicted,
-            convictions: inputs.convictions,
-            doesFlyAirplane: inputs.doesFlyAirplane,
-            doesDriveCar: inputs.doesDriveCar,
-            doesDriveBicycle: inputs.doesDriveBicycle,
+            isConvicted: state.isConvicted,
+            convictions: state.convictions,
+            doesFlyAirplane: state.doesFlyAirplane,
+            doesDriveCar: state.doesDriveCar,
+            doesDriveBicycle: state.doesDriveBicycle,
           }));
         
-      }, [inputs]);
+      }, [state]);
 
       const isDisabled = () => {
         if ( 
-        inputs.doesHaveAgricultureSkills === "" ||
-        (inputs.doesHaveAgricultureSkills === true && !inputs.agricultureSkills) ||
-        inputs.doesHaveMetalworkSkills === "" ||
-        (inputs.doesHaveMetalworkSkills === true && !inputs.metalworkSkills) ||
-        inputs.isConvicted === "" ||
-        (inputs.isConvicted === true && !(inputs.convictions.every(checkConvictions))) ||
-        inputs.doesDriveCar === "" ||
-        inputs.doesFlyAirplane === "" ||
-        inputs.doesDriveBicycle === "" 
+        state.doesHaveAgricultureSkills === "" ||
+        (state.doesHaveAgricultureSkills === true && !state.agricultureSkills) ||
+        state.doesHaveMetalworkSkills === "" ||
+        (state.doesHaveMetalworkSkills === true && !state.metalworkSkills) ||
+        state.isConvicted === "" ||
+        (state.isConvicted === true && !(state.convictions.every(checkConvictions))) ||
+        state.doesDriveCar === "" ||
+        state.doesFlyAirplane === "" ||
+        state.doesDriveBicycle === "" 
         ) 
         { return true;
```

Comment:
Maybe we can just return the condition from the if check? It will evaluate to boolean value. We can have this as a simple variable too, no need for the function. Just a suggestion though :)

Btw, the same goes for the checkConvictions function - we can just return from the function () => (el.forWhat !== "" && el.convictionDate !== "");